extensions/amp-experiment/0.1/amp-experiment.js

```diff
@@ -44,7 +44,13 @@ export class AmpExperiment extends AMP.BaseElement {
     const config = this.getConfig_();
     const results = Object.create(null);
     const variants = Object.keys(config).map(experimentName => {
-      return allocateVariant(this.getWin(), config[experimentName])
+      assertName(experimentName);
+      const experimentConfig = config[experimentName];
+      // If no 'grouping' is specified, fallback to experiment name.
+      if (!experimentConfig.grouping) {
```

Comment:
Mutating an object to meet dependencies is usually considered dirty. We don't share the state to anything but direct dependencies, so it's not horrible to do here, but we can still avoid it.